documentation/reference/ci_cd.md

```diff
@@ -0,0 +1,489 @@
+has not changed, and it simply passes the checks by having the same job names.
+
+### `playwright`
+
+Runs a matrix of various Playwright tests for the frontend using the following
+Node.js scripts.
+
+- `test:playwright visual-regression`
+- `test:playwright e2e`
+- `test:storybook`
+
+This job is skipped if the frontend codebase has not changed.
+
+```{note}
+This job, combined with the [`nuxt-build` job](#nuxt-build), is treated as the
+proof of functionality for deploying frontend to staging.
+```
+
+#### `bypass-playwright`
+
+The `playwright` job is marked as required check in GitHub. If the job is
+skipped, the matrix will not be expanded into individual jobs and all required
+checks will appear as pending, thus blocking PRs from being merged.
+
+To counter this, the `bypass-playwright` job is run if the frontend codebase has
+not changed, and it simply passes the checks by having the same job names.
+
+## Documentation jobs
+
+### `emit-docs`
+
+Builds the documentation and publishes it to an appropriate target. For PRs, the
+target is a subdirectory under `_preview/` of the docs site, so the docs for PR
+#420 will be published at <https://docs.openverse.org/_preview/420/>. For
+commits to `main`, the target is the <https://docs.openverse.org/> site itself.
+
+This job is skipped if neither the documentation nor the frontend codebase has
+changed. It is also not executed on forks and for PRs made by Dependabot.
+
+Documentation is only emitted if all the following conditions are met.
+
+- the event is one of the following
+  - a `push` to `main` and
+    - the repo is `WordPress/openverse` and not a fork
+  - a PR and
+    - the source is a branch of `WordPress/openverse` and not a fork
+    - the author of the PR is not Dependabot
+- either the documentation or the frontend has changed (for the Storybook and
+  Tailwind Config Viewer)
+- all the primary tests ([`test-cat`](#test-cat), [`test-ing`](#test-ing),
+  [`test-api`](#test-api), [`nuxt-build`](#nuxt-build)) succeeded, implying that
+  their changes are valid, or were skipped, implying that they have no changes.
+
+## Docker images jobs
+
+### `publish-images`
+
+Publishes images to GHCR as dictated by the
+[`determine-images`](#determine-images) job. In a matrix, this job downloads the
+images to be published from their `.tar` files using the [`load-img`](#load-img)
+job and then tags and pushes them to GHCR.
+
+Images are only published if all the following conditions are met.
+
+- the event is a push to `main`
+- the repo is `WordPress/openverse` and not a fork
+- there is at least one image to publish as per the `do_publish` output of the
+  [`determine-images`](#determine-images) job
+- all the primary tests ([`test-cat`](#test-cat), [`test-ing`](#test-ing),
+  [`test-api`](#test-api), [`nuxt-build`](#nuxt-build)) succeeded, implying that
+  their changes are valid, or were skipped, implying that they have no changes.
+
+## Deployment jobs
```

Comment:
Do we want to mention in both of these sections that we're deploying only the staging versions of each service, not production? I know that's mentioned at the top but it might be good to have here too.

Reply:
Updated in 6ae49fe.